### PR TITLE
Move Pre and Post Processing Functions to Hooks

### PR DESCRIPTION
**Pull request by dshanske**
This changes the method of removing the generated post content, which I need removed for using Post Kinds, to a removing of a filter over a flag.

At the same time, it allows removing/replacing of upload handling, geodata saving, and mf2 storing to replace with custom functions as needed.

### CODE REVIEW

**Comment by snarfed** [comment]
cool!

have you gotten the tests running? do they pass with this?

**Comment by dshanske** [comment]
I get a Strict Standard warning because you have two extra parameters in the wp_die_handler. And they do pass with this.

File: micropub.php [hunk 1]
```diff
@@ -310,18 +319,12 @@ private static function generate_args() {
         }
       }
     }
-    // If the theme declares it supports microformats2, pass the content through
-    if (current_theme_supports('microformats2')) {
```
Comment (snarfed) [inline micropub.php]: was this removed intentionally? seems like we still want it, right?

File: micropub.php [hunk 2]
```diff
@@ -413,17 +417,23 @@ private static function generate_event() {
...
       require_once(ABSPATH . 'wp-admin/includes/media.php');
       Micropub::check_error(media_handle_upload('photo', $post_id));
     }
+  }
 
+  /**
+   * Adds geodata to post.
+   *
+   */
+  private static function geodata($post_id) {
```
Comment (snarfed) [inline micropub.php]: minor nit: add a verb to this function name? :P

**Comment by snarfed** [comment]
this looks great otherwise, thank you for doing it!

funny also that the parts it touches are the parts we don't have tests for yet. :P maybe consider adding those tests in your next PR!

**Comment by dshanske** [comment]
The add theme support was removed intentionally. The filter removal is cleaner. Because in my case, my plugin is removing it, not a theme. More universal.

**Comment by dshanske** [comment]
I should be able to mirror your tests and try to test for mf2 priorities.

**Comment by dshanske** [comment]
I will push a second commit for the name issue. And will see about testing one or more tonight.

**Comment by snarfed** [comment]
> The add theme support was removed intentionally. The filter removal is cleaner. Because in my case, my plugin is removing it, not a theme. More universal.

hmm. so you mean, plugins and themes that implement/support mf2 should remove `generate_post_content` from the `before_micropub` filter? that means they all need to know the micropub plugin exists, and that they need to do that, which seems error prone. i'm happy with moving it to a filter, but how about we keep the `current_theme_supports` check too?

**Comment by snarfed** [comment]
thanks! you'll need to pull from upstream and rebase this branch and resolve the conflicts, then we can merge!

**Comment by dshanske** [comment]
Didn't notice you'd committed. Sorry.

**Comment by snarfed** [comment]
np, thanks! and nice to see circle running the tests and that they pass too.

mind squashing the commits now?

**Comment by dshanske** [comment]
Will have to wait till tomorrow.

**Comment by dshanske** [comment]
I hope I got that right.

**Comment by snarfed** [comment]
thanks! the squash part looks OK, but this branch isn't rebased onto master cleanly. you can tell when you look at the bottom of the PR, below this comment, there's an error message that says there are conflicts that need to be resolved.

looks like the problem is that you introduced a merge commit, probably when you pulled master into your branch at some point. you don't usually want to do that; you generally want to rebase your branch onto master instead. sounds like you'd looked at some tutorials on this before, maybe check them again?

in the meantime, the easiest way to recover here may be to close this PR, start a new branch, and cherry pick just your one squashed commit.

**Pull request closed without merging**